# XGBOOST

### Importing the libraries

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

### Importing the CANCER DATASET

In [ ]:
dataset = pd.read_csv('/content/Data.csv')
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,-1].values

### Splitting the dataset into Training set and Test Set

In [ ]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=0)

### Training XGBoost on the training set

In [ ]:
from xgboost import XGBClassifier
classifier=XGBClassifier()
classifier.fit(X_train,y_train) # Features of the training set X_train and then the dependent variable of training set y_train


XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, gamma=0,
              learning_rate=0.1, max_delta_step=0, max_depth=3,
              min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,
              nthread=None, objective='binary:logistic', random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,
              silent=None, subsample=1, verbosity=1)

### Making the Confusion Matrix

In [ ]:
from sklearn.metrics import confusion_matrix,accuracy_score
y_pred = classifier.predict(X_test)
cm = confusion_matrix(y_test, y_pred)
print(cm)
accuracy_score(y_test,y_pred)  # Initially the accuracy of the model was 93% but once after XGBoost the accuracy went to 97.8% which is quite impressive

[[84  3]
 [ 0 50]]


0.9781021897810219

### Applying k-Fold Cross Validation

In [ ]:
from sklearn.model_selection import cross_val_score
accuracies = cross_val_score(estimator = classifier, X = X_train, y = y_train, cv = 10)
print("Accuracy:{:.2f} %".format(accuracies.mean()*100))
print("Standard Deviation:{:.2f} %".format(accuracies.std()*100))

Accuracy:96.53 %
Standard Deviation:2.07 %


So the accuracy increased fro 93% to 96.53% which is very good and this is the reason XGBoost is a very powerful library as it increases the model performance.

In [1]:
# When we are dealing with the dataset of the cancer patients its very important to avoid false negatives. Suppose if we predict that the patient does not have cancer but if he actually has cancer then that becomes is a very big problem
# But in our case the number of false negatives are 0 as our model has worked perfectly!